presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -52,6 +53,15 @@ public RuleAssert assertThat(Rule rule)
         return new RuleAssert(metadata, session, rule);
     }
 
+    public void assertInTransaction(Rule rule, Consumer<RuleAssert> ruleAssertion)
```

Comment:
We should make RuleAssert do this, instead. There's no benefit from having two versions.